internal/plugin_test.go

```diff
@@ -2663,3 +2665,164 @@ func TestGenerateNonDNSBindingName(t *testing.T) {
 		})
 	}
 }
+
+func TestConfigTypos(t *testing.T) {
+	t.Parallel()
+	tmpDir := t.TempDir()
+	createConfigMap(t, tmpDir, "configmap.yaml")
+
+	suggestFmt := "line %v: field %v found but not defined in type %v - did you mean '%v'?"
+
+	tests := map[string]struct {
+		desiredErrs []string
+		generator   []byte
+	}{
+		"no typos": {
+			desiredErrs: []string{},
+			generator: []byte(`
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  name: policy-generator-minimal
+policyDefaults:
+  namespace: minimal
+policies:
+- name: my-minimal
+  manifests:
+  - path: @@@
+`),
+		},
+		"one typo with suggestion": {
+			desiredErrs: []string{
+				fmt.Sprintf(suggestFmt, "6", "policyDefault", "PolicyGenerator", "policyDefaults"),
+			},
+			generator: []byte(`
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  name: policy-generator-minimal
+policyDefault:
+  namespace: minimal
+policies:
+- name: my-minimal
+  manifests:
+  - path: @@@
+`),
+		},
+		"two typos with suggestions": {
+			desiredErrs: []string{
+				fmt.Sprintf(suggestFmt, "6", "policyDefault", "PolicyGenerator", "policyDefaults"),
+				fmt.Sprintf(suggestFmt, "8", "policie", "PolicyGenerator", "policies"),
+			},
+			generator: []byte(`
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  name: policy-generator-minimal
+policyDefault:
+  namespace: minimal
+policie:
+- name: my-minimal
+  manifests:
+  - path: @@@
+`),
+		},
+		"one deeper typo": {
+			desiredErrs: []string{
+				fmt.Sprintf(suggestFmt, "7", "configPolicyAnnotations", "policyDefaults",
+					"configurationPolicyAnnotations"),
+			},
+			generator: []byte(`
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  name: policy-generator-minimal
+policyDefaults:
+  configPolicyAnnotations: {}
+  namespace: minimal
+policies:
+- name: my-minimal
+  manifests:
+  - path: @@@
+`),
+		},
+		"typo inside a list": {
+			desiredErrs: []string{
+				fmt.Sprintf(suggestFmt, "11", "paths", "manifests", "path"),
+			},
+			generator: []byte(`
+apiVersion: policy.open-cluster-management.io/v1
+kind: PolicyGenerator
+metadata:
+  name: policy-generator-minimal
+policyDefaults:
+  namespace: minimal
+policies:
+- name: my-minimal
+  manifests:
+  - paths: @@@
+`),
+		},
+		"typo no suggestion": {
+			desiredErrs: []string{
+				"line 12: field namespace found but not defined in type manifests$",
```

Comment:
Not an issue and more of a curiosity question: What's the meaning of the trailing `$`?